Xamarin.PropertyEditing.Mac/Controls/RequestResourceView.cs

```diff
@@ -0,0 +1,507 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using Xamarin.PropertyEditing.Drawing;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class ResourceTableDataSource
+		: NSTableViewDataSource
+	{
+		private ResourceSelectorViewModel viewModel;
+		internal ResourceTableDataSource (ResourceSelectorViewModel propertyViewModel)
+		{
+			if (propertyViewModel == null)
+				throw new ArgumentNullException (nameof (propertyViewModel));
+
+			this.viewModel = propertyViewModel;
+		}
+
+		public ResourceSelectorViewModel ViewModel => this.viewModel;
+
+		public override nint GetRowCount (NSTableView tableView)
+		{
+			return this.viewModel.Resources.Count ();
```

Comment:
We can convert this to the property when it's based on #339

Reply:
Done.